src/storage/s3.rs

```diff
@@ -35,6 +35,20 @@ impl S3Backend {
         }
     }
 
+    pub(super) fn exists(&self, path: &str) -> Result<bool, Error> {
+        S3_RUNTIME.handle().block_on(async {
+            let resp = self
+                .client
+                .list_objects_v2(ListObjectsV2Request {
```

Comment:
A list request is ~10x more expensive money wise than a get request.

Reply:
I changed it to use HEAD instead.